File: java-integration/src/main/java/br/com/acqio/enums/CardApplication.java
package br.com.acqio.enums;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public enum CardApplication {
    CREDITO("01", "Crédito"),
    DEBITO("02", "Débito");

    private final String code;
    private final String application;

}

File: csharp/csharp/enums/CardApplication.cs
namespace csharp.dto
{
    enum CardApplication
    {
        CREDITO,
        DEBITO
    }

    static class CardApplicationExtension
    {
        internal static string GetDescription(CardApplication cardApplication)
        {
            if (CardApplication.CREDITO == cardApplication)
            {
                return "Crédito";
            }
            else
            {
                return "Débito";
            }
        }


        internal static string GetCode(CardApplication cardApplication)
        {
            if (CardApplication.CREDITO == cardApplication)
            {
                return "01";
            }
            else
            {
                return "02";
            }
        }

        internal static CardApplication GetCardApplication(string cardApplication)
        {
            if (cardApplication == "01" || cardApplication == "Crédito")
            {
                return CardApplication.CREDITO;
            }
            else
            {
                return CardApplication.DEBITO;
            }
        }

    }

}